0.3.0-alpha.2 segmentation violation code=0x1 addr=0x0 pc=0xf4bc4a

/kind bug

**What happened**: I wanted to test cert-manager 0.3.0 alpha, just started it up and it got a nil pointer exception

**What you expected to happen**: It not getting a nil pointer exception and it creating the cert via the http challenge

**How to reproduce it (as minimally and precisely as possible)**:
I upgraded from 0.2.5 but first deleted the clusterissues, deployment, secrets, certificate etc to have a fresh start.

**Anything else we need to know?**:

It did manage to create the new ingress resource which is pointing to the acme challenge. 

Logs from cert-manager:
```
I0503 08:59:54.368667       1 server.go:68] Listening on http://0.0.0.0:9402
I0503 08:59:54.369960       1 controller.go:102] Creating event broadcaster
I0503 08:59:54.370595       1 leaderelection.go:175] attempting to acquire leader lease  kube-system/cert-manager-controller...
I0503 08:59:54.466695       1 request.go:481] Throttling request took 96.027248ms, request: GET:https://100.64.0.1:443/api/v1/namespaces/kube-system/configmaps/cert-manager-controller
I0503 08:59:54.476516       1 round_trippers.go:405] GET https://100.64.0.1:443/api/v1/namespaces/kube-system/configmaps/cert-manager-controller 200 OK in 9 milliseconds
I0503 08:59:54.566760       1 round_trippers.go:405] PUT https://100.64.0.1:443/api/v1/namespaces/kube-system/configmaps/cert-manager-controller 200 OK in 88 milliseconds
I0503 08:59:54.567004       1 leaderelection.go:184] successfully acquired lease kube-system/cert-manager-controller
I0503 08:59:54.567333       1 controller.go:54] Starting shared informer factory
I0503 08:59:54.567511       1 controller.go:104] Event(v1.ObjectReference{Kind:"ConfigMap", Namespace:"kube-system", Name:"cert-manager-controller", UID:"d95ad54c-4eae-11e8-8a82-060ce275168e", APIVersion:"v1", ResourceVersion:"1748466", FieldPath:""}): type: 'Normal' reason: 'LeaderElection' cert-manager-kube-system-certmanager-6f5669bb6f-m8ktz-external-cert-manager-controller became leader
I0503 08:59:54.567551       1 controller.go:45] Starting issuers controller
I0503 08:59:54.567561       1 controller.go:96] Starting issuers control loop
I0503 08:59:54.567594       1 controller.go:45] Starting certificates controller
I0503 08:59:54.567601       1 controller.go:136] Starting certificates control loop
I0503 08:59:54.567626       1 controller.go:45] Starting clusterissuers controller
I0503 08:59:54.567647       1 controller.go:98] Starting clusterissuers control loop
I0503 08:59:54.567672       1 controller.go:45] Starting ingress-shim controller
I0503 08:59:54.567678       1 controller.go:111] Starting ingress-shim control loop
I0503 08:59:54.567866       1 reflector.go:202] Starting reflector *v1.Pod (30s) from github.com/jetstack/cert-manager/vendor/k8s.io/client-go/informers/factory.go:87
I0503 08:59:54.567889       1 reflector.go:240] Listing and watching *v1.Pod from github.com/jetstack/cert-manager/vendor/k8s.io/client-go/informers/factory.go:87
I0503 08:59:54.568193       1 reflector.go:202] Starting reflector *v1alpha1.Issuer (30s) from github.com/jetstack/cert-manager/pkg/client/informers/externalversions/factory.go:71
I0503 08:59:54.568205       1 reflector.go:240] Listing and watching *v1alpha1.Issuer from github.com/jetstack/cert-manager/pkg/client/informers/externalversions/factory.go:71
I0503 08:59:54.568481       1 reflector.go:202] Starting reflector *v1alpha1.ClusterIssuer (30s) from github.com/jetstack/cert-manager/pkg/client/informers/externalversions/factory.go:71
I0503 08:59:54.568493       1 reflector.go:240] Listing and watching *v1alpha1.ClusterIssuer from github.com/jetstack/cert-manager/pkg/client/informers/externalversions/factory.go:71
I0503 08:59:54.568846       1 reflector.go:202] Starting reflector *v1alpha1.Certificate (30s) from github.com/jetstack/cert-manager/pkg/client/informers/externalversions/factory.go:71
I0503 08:59:54.568861       1 reflector.go:240] Listing and watching *v1alpha1.Certificate from github.com/jetstack/cert-manager/pkg/client/informers/externalversions/factory.go:71
I0503 08:59:54.569171       1 reflector.go:202] Starting reflector *v1.Secret (30s) from github.com/jetstack/cert-manager/vendor/k8s.io/client-go/informers/factory.go:87
I0503 08:59:54.569182       1 reflector.go:240] Listing and watching *v1.Secret from github.com/jetstack/cert-manager/vendor/k8s.io/client-go/informers/factory.go:87
I0503 08:59:54.569447       1 reflector.go:202] Starting reflector *v1.Service (30s) from github.com/jetstack/cert-manager/vendor/k8s.io/client-go/informers/factory.go:87
I0503 08:59:54.569460       1 reflector.go:240] Listing and watching *v1.Service from github.com/jetstack/cert-manager/vendor/k8s.io/client-go/informers/factory.go:87
I0503 08:59:54.666859       1 reflector.go:202] Starting reflector *v1beta1.Ingress (30s) from github.com/jetstack/cert-manager/vendor/k8s.io/client-go/informers/factory.go:87
I0503 08:59:54.666898       1 reflector.go:240] Listing and watching *v1beta1.Ingress from github.com/jetstack/cert-manager/vendor/k8s.io/client-go/informers/factory.go:87
I0503 08:59:54.667052       1 round_trippers.go:405] GET https://100.64.0.1:443/api/v1/services?limit=500&resourceVersion=0 200 OK in 97 milliseconds
I0503 08:59:54.667084       1 round_trippers.go:405] GET https://100.64.0.1:443/apis/certmanager.k8s.io/v1alpha1/issuers?limit=500&resourceVersion=0 200 OK in 98 milliseconds
I0503 08:59:54.667100       1 round_trippers.go:405] GET https://100.64.0.1:443/apis/certmanager.k8s.io/v1alpha1/clusterissuers?limit=500&resourceVersion=0 200 OK in 98 milliseconds
I0503 08:59:54.667113       1 round_trippers.go:405] GET https://100.64.0.1:443/apis/certmanager.k8s.io/v1alpha1/certificates?limit=500&resourceVersion=0 200 OK in 98 milliseconds
I0503 08:59:54.667172       1 round_trippers.go:405] GET https://100.64.0.1:443/api/v1/secrets?limit=500&resourceVersion=0 200 OK in 97 milliseconds
I0503 08:59:54.668107       1 round_trippers.go:405] POST https://100.64.0.1:443/api/v1/namespaces/kube-system/events 201 Created in 100 milliseconds
I0503 08:59:54.768910       1 round_trippers.go:405] GET https://100.64.0.1:443/api/v1/pods?limit=500&resourceVersion=0 200 OK in 200 milliseconds
I0503 08:59:54.866789       1 round_trippers.go:405] GET https://100.64.0.1:443/apis/extensions/v1beta1/ingresses?limit=500&resourceVersion=0 200 OK in 199 milliseconds
I0503 08:59:55.499170       1 shared_informer.go:123] caches populated
I0503 08:59:55.513690       1 controller.go:119] Starting "issuers" worker
I0503 08:59:55.516708       1 shared_informer.go:123] caches populated
I0503 08:59:55.516812       1 controller.go:117] Synced all caches for ingress-shim control loop
I0503 08:59:55.516914       1 shared_informer.go:123] caches populated
I0503 08:59:55.570965       1 round_trippers.go:405] GET https://100.64.0.1:443/apis/certmanager.k8s.io/v1alpha1/certificates?resourceVersion=1747875&timeoutSeconds=514&watch=true 200 OK in 302 milliseconds
I0503 08:59:55.576754       1 controller.go:119] Starting "issuers" worker
I0503 08:59:55.602107       1 round_trippers.go:405] GET https://100.64.0.1:443/apis/extensions/v1beta1/ingresses?resourceVersion=1744527&timeoutSeconds=582&watch=true 200 OK in 732 milliseconds
I0503 08:59:55.602590       1 round_trippers.go:405] GET https://100.64.0.1:443/apis/certmanager.k8s.io/v1alpha1/issuers?resourceVersion=1282116&timeoutSeconds=360&watch=true 200 OK in 732 milliseconds
I0503 08:59:55.602700       1 round_trippers.go:405] GET https://100.64.0.1:443/apis/certmanager.k8s.io/v1alpha1/clusterissuers?resourceVersion=1747991&timeoutSeconds=359&watch=true 200 OK in 435 milliseconds
I0503 08:59:55.602852       1 round_trippers.go:405] GET https://100.64.0.1:443/api/v1/secrets?resourceVersion=1747941&timeoutSeconds=320&watch=true 200 OK in 433 milliseconds
I0503 08:59:55.602874       1 controller.go:119] Starting "issuers" worker
I0503 08:59:55.602886       1 controller.go:119] Starting "issuers" worker
I0503 08:59:55.602897       1 controller.go:119] Starting "issuers" worker
I0503 08:59:55.603165       1 controller.go:121] Starting "clusterissuers" worker
I0503 08:59:55.713341       1 controller.go:138] clusterissuers controller: syncing item 'letsencrypt-staging'
I0503 08:59:55.715070       1 acme.go:159] getting private key (letsencrypt-staging->tls.key) for acme issuer kube-system/letsencrypt-staging
I0503 08:59:55.716462       1 logger.go:67] Calling GetAccount
I0503 08:59:55.603182       1 controller.go:135] Starting "ingress-shim" worker
I0503 08:59:55.603192       1 controller.go:121] Starting "clusterissuers" worker
I0503 08:59:55.603201       1 controller.go:121] Starting "clusterissuers" worker
I0503 08:59:55.603210       1 controller.go:121] Starting "clusterissuers" worker
I0503 08:59:55.603218       1 controller.go:121] Starting "clusterissuers" worker
I0503 08:59:55.603227       1 controller.go:135] Starting "ingress-shim" worker
I0503 08:59:55.603236       1 controller.go:135] Starting "ingress-shim" worker
I0503 08:59:55.603244       1 controller.go:135] Starting "ingress-shim" worker
I0503 08:59:55.603253       1 controller.go:135] Starting "ingress-shim" worker
I0503 08:59:55.666874       1 round_trippers.go:405] GET https://100.64.0.1:443/api/v1/services?resourceVersion=1744409&timeoutSeconds=366&watch=true 200 OK in 56 milliseconds
I0503 08:59:55.767840       1 controller.go:152] ingress-shim controller: syncing item 'prod/api-proxy'
I0503 08:59:55.768102       1 sync.go:123] Certificate "api-proxy-tls" for ingress "api-proxy" already exists
I0503 08:59:55.768111       1 sync.go:126] Certificate "api-proxy-tls" for ingress "api-proxy" is up to date
I0503 08:59:55.768125       1 controller.go:166] ingress-shim controller: Finished processing work item "prod/api-proxy"
I0503 08:59:55.767990       1 controller.go:138] clusterissuers controller: syncing item 'letsencrypt-production'
I0503 08:59:55.768163       1 acme.go:159] getting private key (letsencrypt-production->tls.key) for acme issuer kube-system/letsencrypt-production
I0503 08:59:55.770087       1 logger.go:67] Calling GetAccount
I0503 08:59:55.768022       1 controller.go:152] ingress-shim controller: syncing item 'prod/cm-acme-http-solver-2xvnd'
I0503 08:59:55.770315       1 sync.go:48] Not syncing ingress prod/cm-acme-http-solver-2xvnd as it does not contain necessary annotations
I0503 08:59:55.770330       1 controller.go:166] ingress-shim controller: Finished processing work item "prod/cm-acme-http-solver-2xvnd"
I0503 08:59:56.766842       1 round_trippers.go:405] GET https://100.64.0.1:443/api/v1/namespaces/kube-system/configmaps/cert-manager-controller 200 OK in 199 milliseconds
I0503 08:59:56.770963       1 round_trippers.go:405] PUT https://100.64.0.1:443/api/v1/namespaces/kube-system/configmaps/cert-manager-controller 200 OK in 3 milliseconds
I0503 08:59:56.771224       1 leaderelection.go:199] successfully renewed lease kube-system/cert-manager-controller
I0503 08:59:57.867535       1 shared_informer.go:123] caches populated
I0503 08:59:57.867755       1 controller.go:142] Synced all caches for certificates control loop
I0503 08:59:57.867839       1 controller.go:160] Starting "certificates" worker
I0503 08:59:57.867854       1 controller.go:160] Starting "certificates" worker
I0503 08:59:57.867842       1 controller.go:160] Starting "certificates" worker
I0503 08:59:57.867929       1 controller.go:177] certificates controller: syncing item 'prod/api-proxy-tls'
I0503 08:59:57.867965       1 controller.go:160] Starting "certificates" worker
I0503 08:59:57.868060       1 sync.go:239] Preparing certificate prod/api-proxy-tls with issuer
I0503 08:59:57.868070       1 prepare.go:47] Getting ACME client
I0503 08:59:57.868076       1 acme.go:159] getting private key (letsencrypt-staging->tls.key) for acme issuer kube-system/letsencrypt-staging
I0503 08:59:57.867930       1 controller.go:160] Starting "certificates" worker
I0503 08:59:57.868453       1 round_trippers.go:405] GET https://100.64.0.1:443/api/v1/pods?resourceVersion=1748460&timeoutSeconds=509&watch=true 200 OK in 1 milliseconds
I0503 08:59:57.868543       1 logger.go:27] Calling GetOrder
I0503 08:59:58.869664       1 round_trippers.go:405] GET https://100.64.0.1:443/api/v1/namespaces/kube-system/configmaps/cert-manager-controller 200 OK in 2 milliseconds
I0503 08:59:58.874052       1 round_trippers.go:405] PUT https://100.64.0.1:443/api/v1/namespaces/kube-system/configmaps/cert-manager-controller 200 OK in 4 milliseconds
I0503 08:59:58.966788       1 leaderelection.go:199] successfully renewed lease kube-system/cert-manager-controller
I0503 09:00:00.970235       1 round_trippers.go:405] GET https://100.64.0.1:443/api/v1/namespaces/kube-system/configmaps/cert-manager-controller 200 OK in 2 milliseconds
I0503 09:00:00.974501       1 round_trippers.go:405] PUT https://100.64.0.1:443/api/v1/namespaces/kube-system/configmaps/cert-manager-controller 200 OK in 4 milliseconds
I0503 09:00:00.974594       1 leaderelection.go:199] successfully renewed lease kube-system/cert-manager-controller
E0503 09:00:01.341202       1 sync.go:168] [prod/api-proxy-tls] Error getting certificate 'api-proxy-tls': secret "api-proxy-tls" not found
E0503 09:00:01.341321       1 runtime.go:66] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:509
/usr/local/go/src/runtime/panic.go:491
/usr/local/go/src/runtime/panic.go:63
/usr/local/go/src/runtime/signal_unix.go:367
/go/src/github.com/jetstack/cert-manager/third_party/crypto/acme/types.go:67
/go/src/github.com/jetstack/cert-manager/pkg/issuer/acme/prepare.go:447
/go/src/github.com/jetstack/cert-manager/pkg/issuer/acme/prepare.go:58
/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/sync.go:240
/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/sync.go:108
/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:215
/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:69
/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:178
/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:183
/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:147
/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:2337
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xf4bc4a]

goroutine 124 [running]:
github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x111
panic(0x1385860, 0x1efd7e0)
	/usr/local/go/src/runtime/panic.go:491 +0x283
github.com/jetstack/cert-manager/third_party/crypto/acme.(*Error).Error(0x0, 0x15248f1, 0xe)
	/go/src/github.com/jetstack/cert-manager/third_party/crypto/acme/types.go:67 +0x4a
github.com/jetstack/cert-manager/pkg/issuer/acme.(*Acme).shouldAttemptValidation(0xc420392600, 0x1ea8420, 0xc420867900, 0x1eb0140, 0xc420222f30, 0xc42005e1a0, 0xa0, 0x34, 0x400, 0x34)
	/go/src/github.com/jetstack/cert-manager/pkg/issuer/acme/prepare.go:447 +0x390
github.com/jetstack/cert-manager/pkg/issuer/acme.(*Acme).Prepare(0xc420392600, 0x1ea8420, 0xc420867900, 0xc42005e1a0, 0x2, 0xc4208af998)
	/go/src/github.com/jetstack/cert-manager/pkg/issuer/acme/prepare.go:58 +0x230
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).issue(0xc4203ca000, 0x1ea8420, 0xc420867900, 0x1ea86e0, 0xc420392600, 0xc42005e1a0, 0x0, 0x1e8b8a0)
	/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/sync.go:240 +0x163
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).Sync(0xc4203ca000, 0x1ea8420, 0xc420867900, 0xc420066a00, 0x1e8b8a0, 0xc420302990)
	/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/sync.go:108 +0x385
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).processNextWorkItem(0xc4203ca000, 0x1ea8420, 0xc420867900, 0xc4202c9a60, 0x12, 0x2, 0x2)
	/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:215 +0x2b5
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).(github.com/jetstack/cert-manager/pkg/controller/certificates.processNextWorkItem)-fm(0x1ea8420, 0xc420867900, 0xc4202c9a60, 0x12, 0x2, 0x130c6c0)
	/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:69 +0x52
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).worker.func1(0xc4203ca000, 0xc420376da8, 0xc42005c7e0, 0x130c6c0, 0xc420210810, 0x0, 0x0)
	/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:178 +0x24f
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).worker(0xc4203ca000, 0xc42005c7e0)
	/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:183 +0x166
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).Run.func1()
	/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:147 +0x33
github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc4207e6ca0)
	/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x5e
github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc4207e6ca0, 0x3b9aca00, 0x0, 0x1, 0xc42005c7e0)
	/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc4207e6ca0, 0x3b9aca00, 0xc42005c7e0)
	/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).Run
	/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:147 +0x182
```

**Environment**:
- Kubernetes version (use `kubectl version`): `1.9.6`
- Cloud provider or hardware configuration**: AWS managed by Kops

Same / similar issue for us:


```
I0504 06:10:37.787712       1 controller.go:177] certificates controller: syncing item 'monitoring/my-certificate-name'
I0504 06:10:37.787883       1 sync.go:239] Preparing certificate monitoring/my-certificate-name with issuer
I0504 06:10:37.787928       1 acme.go:159] getting private key (letsencrypt->tls.key) for acme issuer cert-manager/letsencrypt-cluster-issuer-private
I0504 06:10:37.788540       1 logger.go:27] Calling GetOrder
I0504 06:10:38.008802       1 logger.go:52] Calling GetAuthorization
I0504 06:10:38.195145       1 logger.go:77] Calling DNS01ChallengeRecord
I0504 06:10:38.195179       1 prepare.go:263] Cleaning up old/expired challenges for Certificate monitoring/my-certificate-name
I0504 06:10:38.195192       1 logger.go:47] Calling GetChallenge
I0504 06:10:38.373619       1 dns.go:78] Checking DNS propagation for "my.fqdn.domain" using name servers: [10.200.96.96:53]
I0504 06:10:38.460149       1 dns.go:89] Waiting DNS record TTL (60s) to allow propagation of DNS record for domain "_acme-challenge.my.fqdn.domain."
I0504 06:11:38.460359       1 dns.go:91] ACME DNS01 validation record propagated for "_acme-challenge.my.fqdn.domain."
I0504 06:11:38.460390       1 prepare.go:468] Accepting challenge for domain "my.fqdn.domain"
I0504 06:11:38.460424       1 logger.go:42] Calling AcceptChallenge
I0504 06:11:39.183734       1 prepare.go:480] Waiting for authorization for domain "my.fqdn.domain"
I0504 06:11:39.183756       1 logger.go:57] Calling WaitAuthorization
I0504 06:11:41.032401       1 helpers.go:162] Found status change for Certificate "my-certificate-name" condition "Ready": "False" -> "False"; setting lastTransitionTime to 2018-05-04 06:11:41.03238981 +0000 UTC m=+1120.315200364
I0504 06:11:41.032452       1 sync.go:241] Error preparing issuer for certificate monitoring/my-certificate-name: acme: authorization for identifier my.fqdn.domain is invalid
E0504 06:11:41.036987       1 sync.go:168] [monitoring/my-certificate-name] Error getting certificate 'tls-my-certificate-name': secret "tls-my-certificate-name" not found
E0504 06:11:41.037009       1 controller.go:186] certificates controller: Re-queuing item "monitoring/my-certificate-name" due to error processing: acme: authorization for identifier my.fqdn.domain is invalid
I0504 06:12:41.036701       1 controller.go:177] certificates controller: syncing item 'monitoring/my-certificate-name'
I0504 06:12:41.036813       1 sync.go:239] Preparing certificate monitoring/my-certificate-name with issuer
I0504 06:12:41.036837       1 acme.go:159] getting private key (letsencrypt->tls.key) for acme issuer cert-manager/letsencrypt-cluster-issuer-private
I0504 06:12:41.037154       1 logger.go:27] Calling GetOrder
E0504 06:12:41.249541       1 sync.go:168] [monitoring/my-certificate-name] Error getting certificate 'tls-my-certificate-name': secret "tls-my-certificate-name" not found
E0504 06:12:41.249720       1 runtime.go:66] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:509
/usr/local/go/src/runtime/panic.go:491
/usr/local/go/src/runtime/panic.go:63
/usr/local/go/src/runtime/signal_unix.go:367
/go/src/github.com/jetstack/cert-manager/third_party/crypto/acme/types.go:67
/go/src/github.com/jetstack/cert-manager/pkg/issuer/acme/prepare.go:447
/go/src/github.com/jetstack/cert-manager/pkg/issuer/acme/prepare.go:58
/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/sync.go:240
/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/sync.go:108
/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:215
/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:69
/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:178
/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:183
/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:147
/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:2337
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xf4bc4a]

goroutine 197 [running]:
github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x111
panic(0x1385860, 0x1efd7e0)
	/usr/local/go/src/runtime/panic.go:491 +0x283
github.com/jetstack/cert-manager/third_party/crypto/acme.(*Error).Error(0x0, 0x15248f1, 0xe)
	/go/src/github.com/jetstack/cert-manager/third_party/crypto/acme/types.go:67 +0x4a
github.com/jetstack/cert-manager/pkg/issuer/acme.(*Acme).shouldAttemptValidation(0xc420010c00, 0x1ea8420, 0xc42004c3c0, 0x1eb0140, 0xc4202b8510, 0xc42027eea0, 0x24f, 0x56, 0x60, 0x56)
	/go/src/github.com/jetstack/cert-manager/pkg/issuer/acme/prepare.go:447 +0x390
github.com/jetstack/cert-manager/pkg/issuer/acme.(*Acme).Prepare(0xc420010c00, 0x1ea8420, 0xc42004c3c0, 0xc42027eea0, 0x2, 0xc4206cb998)
	/go/src/github.com/jetstack/cert-manager/pkg/issuer/acme/prepare.go:58 +0x230
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).issue(0xc4203f4000, 0x1ea8420, 0xc42004c3c0, 0x1ea86e0, 0xc420010c00, 0xc42027eea0, 0x0, 0x1e8b8a0)
	/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/sync.go:240 +0x163
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).Sync(0xc4203f4000, 0x1ea8420, 0xc42004c3c0, 0xc420076340, 0x1e8b8a0, 0xc420772090)
	/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/sync.go:108 +0x385
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).processNextWorkItem(0xc4203f4000, 0x1ea8420, 0xc42004c3c0, 0xc420bbd340, 0x34, 0x2, 0x2)
	/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:215 +0x2b5
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).(github.com/jetstack/cert-manager/pkg/controller/certificates.processNextWorkItem)-fm(0x1ea8420, 0xc42004c3c0, 0xc420bbd340, 0x34, 0x2, 0x130c6c0)
	/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:69 +0x52
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).worker.func1(0xc4203f4000, 0xc420677da8, 0xc42006e840, 0x130c6c0, 0xc4200431b0, 0x0, 0x0)
	/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:178 +0x24f
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).worker(0xc4203f4000, 0xc42006e840)
	/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:183 +0x166
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).Run.func1()
	/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:147 +0x33
github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc420576660)
	/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x5e
github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc420576660, 0x3b9aca00, 0x0, 0x1, 0xc42006e840)
	/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc420576660, 0x3b9aca00, 0xc42006e840)
	/go/src/github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).Run
	/go/src/github.com/jetstack/cert-manager/pkg/controller/certificates/controller.go:147 +0x182
```

**Environment**:
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.1", GitCommit:"d4ab47518836c750f9949b9e0d387f20fb92260b", GitTreeState:"clean", BuildDate:"2018-04-13T22:29:03Z", GoVersion:"go1.9.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.2", GitCommit:"81753b10df112992bf51bbc2c2f85208aad78335", GitTreeState:"clean", BuildDate:"2018-04-27T09:10:24Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:

Kubernetes cluster on Azure, deployed with acs-engine

--- a/pkg/issuer/acme/prepare.go
+++ b/pkg/issuer/acme/prepare.go
@@ -444,7 +444,11 @@ func (a *Acme) shouldAttemptValidation(ctx context.Context, cl client.Interface,
 			Type:   v1alpha1.CertificateConditionValidationFailed,
 			Status: v1alpha1.ConditionTrue,
 		}) {
-			crt.UpdateStatusCondition(v1alpha1.CertificateConditionValidationFailed, v1alpha1.ConditionTrue, "OrderFailed", fmt.Sprintf("Order failed: %v", order.Error.Error()), true)
+			var extraText = ""
+			if order.Error != nil {
+				extraText = fmt.Sprintf(": %v", order.Error.Error())
+			}
+			crt.UpdateStatusCondition(v1alpha1.CertificateConditionValidationFailed, v1alpha1.ConditionTrue, "OrderFailed", "Order status is invalid"+extraText, true)
 		}
 
 		// we know that we'll be able to find the appropriate condition because

